0017-ilp-errors/0017-ilp-errors.md

```diff
@@ -0,0 +1,73 @@
+# Interledger Protocol Errors
+
+This spec defines a standard error format and codes to be used when ILP payments are rejected.
+
+Inspired by [HTTP Status Codes](https://tools.ietf.org/html/rfc2616#section-10), ILP errors are categorized based on the intended behavior of the caller when they get the given error.
+
+## Error Propagation
+
+ILP connectors MUST relay errors when they are notified of a transfer being rejected.
+
+Connectors SHOULD NOT modify the error codes or messages except in the following way: connectors MAY append their ILP addresses, separated from the error message by a newline (`\n`), to assist with debugging/tracing.
+
+## Data Format
+
+ILP errors are UTF-8 encoded strings that are a maximum of 65535 bytes long. Error messages that are too long for a ledger to handle SHOULD be truncated to the first 65535 bytes.
+
+Errors consist of a three character code, followed by a colon (`:`) and an arbitrary message set by the entity that raised the error. Transport and application layer protocols MAY define error messages that have special significance in their context.
```

Comment:
I think both of those would be nice to have.

This makes me think we should use the HTTP data format for errors, so it would be something like:
```http
ILP/1.0 S00: Bad Request
Triggered-By: example.ledger.bob
Forwarded-By: example.connector.connie,example.connector.chloe
Date: 2017-03-23T00:00:00Z
Content-Type: application/json

{
  "id": "SomeApplicationError",
  "message": "Oops..."
}
```

Thoughts?